### ISSUE TITLE
PURE(BLUS30182) character shadows broken when in the air

### ISSUE THREAD

**Issue by JimScript** [RPCS3/rpcs3#13684]
### Quick summary

When you are high in the air, your shadow is rendered as a square(it's fine on the ground).

### Details

Here is a video of it in action:

https://user-images.githubusercontent.com/66685584/232347790-18e84f68-1ef0-405a-a756-89c6bd7ae190.mp4

### Attach a log file

Log File:[RPCS3_PURE.log](https://github.com/RPCS3/rpcs3/files/11244875/RPCS3_PURE.log)

### Attach capture files for visual issues

RSX Capture:[BLUS30182_20230416151022_capture.zip](https://github.com/RPCS3/rpcs3/files/11244878/BLUS30182_20230416151022_capture.zip)

### System configuration

- SYS: AMD Ryzen 5 3600 6-Core Processor | 12 Threads | 15.95 GiB RAM | TSC: 3.600GHz | AVX+ | FMA3
- SYS: Operating system: Windows, Major: 10, Minor: 0, Build: 19045, Service Pack: none, Compatibility mode: 0
- RSX: Found vulkan-compatible GPU: 'NVIDIA GeForce RTX 2070 SUPER' running on driver 522.25.0.0
- SYS: Firmware version: 4.88

**Comment by JimScript** [comment]
Right, well I checked to see if this was a regression, but when I "found" it it was when the shader precision setting was added. So I set it to "Low" and it fixed it, so I'm closing this as it is not an issue.

**Comment by JimScript** [comment]
I guess I didn't test it enough, the "Low" Shader Precision fix only works on my Steam Deck, my main computer still shows a square. So I guess it is a regression(#12788), but NVIDIA/Windows seems to be worse off since that fix doesn't work.

**Comment by alexconsole04** [comment]
1 year late, but this issue still persists on latest versions. Found a video from 2017 and it did show the same issue. While it was taken during when the game was likely labeled Ingame at the time, it still shows the exact same rendering issues with shadows. https://youtu.be/8OKXhWHlOYw?si=RiVNvj_yW5h4Byjx

But essentially, shadows get much more screwed up with other racers as seen on this race demonstration that is on the latest version of RPCS3 
https://youtu.be/7tQzAQZCT04. 

I'm very uncertain this is a GPU related issue, but that's an conclusion that came up to me while doing research and watching various other gameplay videos of this game on YT. No settings from what it seems fixes this issue, even with all the Buffers enabled.

My log
[RPCS3.log.gz](https://github.com/user-attachments/files/18147073/RPCS3.log.gz)

**Comment by JimScript** [comment]
It's not any different...

**Comment by digant73** [comment]
@kd-11 this issue is a regression since 0.0.21-13531 (#11867), not from the reported PR #12788.
@JimScript please check from your side
You can use the Pure demo version for testing (just in the initial trial)

### 0.0.21-13528 - OK

<img width="2889" height="1685" alt="Image" src="https://github.com/user-attachments/assets/17b8becf-39fe-4110-a60d-662d041fb07f" />

### 0.0.21-13531 (#11867) - KO

<img width="2879" height="1658" alt="Image" src="https://github.com/user-attachments/assets/d9604f0d-5d92-408c-8a3d-fd040ade39f6" />

**Comment by digant73** [comment]
Follow up of the regressing PR #11867.

Rolling back the changes made by #11867 on current build 0.0.41-19453 (that in the meanwhile added `Shader Quality` setting), the issue is not present when using `Shade Quality` set to `Low` and replacing the following property in `VKVertexProgram.cpp` (a similar property is present for OpenGL) added by the regressing PR:

`properties2.low_precision_tests = vk::is_NVIDIA(vk::get_driver_vendor());`

with:

`properties2.low_precision_tests = false;`

Hopefully the analysis is enough to find and fix the issue

**NOTE:** This issue is also related to #16488 affected by the same regressing PR. So a fix will close both issues
